Add gedi_subsetter H5DataFrame and h5coro methods to template notebook

weiji14 opened the pull request:
Main change is adding the H5DataFrame read in method from #5 into the `benchmark-format-template.ipynb` notebook. Also pulling in the h5coro method from the `benchmark-original-hdf5.ipynb` notebook at c9ac2100e38253334dba2f0e1b7d5a359a6be8d0, and added tqdm progress bars.

weiji14 committed:
Concatenate arrays in h5py to match shape from xarray.combine_by_coords

Ensure that both array shapes from the h5py and xarray read methods are equal to (298271231,)  before computing the mean.

weiji14 commented on notebooks/benchmark-format-template.ipynb:
```diff
@@ -163,7 +165,9 @@
    "source": [
-    "final_array"
+    "%%time\n",
+    "assert final_array.shape == (298271231,)\n",
+    "final_array.mean()"
```
@abarciauskas-bgse, were you thinking of concatenating the `h_ph` (photon height) data variable across all 5 ATL03 files into a single 1D array just like xarray? Or some other shape?

weiji14 replied:
Ah ok, I see that your new commit at c9ac2100e38253334dba2f0e1b7d5a359a6be8d0, let me reconcile those together later (after lunch).

abarciauskas-bgse replied:
Thanks @weiji14 I was just about to comment about that!

weiji14 replied:
Ok, have synced up with the stuff from commit e9aedc4905a85d2349a3c272cf4f6b8399089877, going to squash merge this into main now!

weiji14 committed:
Add H5DataFrame to list of comparison libraries

Added an Option 1b (H5DataFrame) which is still based on h5py, but wraps around pandas.DataFrame. Also putting the data read-in and mean computation in the same cell to have just one timing to compare to.

weiji14 committed:
Merge branch 'main' into match-h5py-xarray-shape

weiji14 committed:
Update benchmark-format-template with h5coro and sync with main

Include h5coro benchmarking code, sync changes with main branch at e9aedc4905a85d2349a3c272cf4f6b8399089877, and also added tqdm progress bar to get an estimate of how long it will take to run.
